server/src/main/java/org/opensearch/index/translog/TranslogCheckedContainer.java

```diff
@@ -28,21 +25,15 @@ public class TranslogCheckedContainer {
     private final Checksum checksum;
     private final AtomicLong contentLength;
     private final ReleasableLock updateLock = new ReleasableLock(new ReentrantLock());
-    private final String file;
 
     /**
-     * Creates TranslogCheckedContainer from provided channel.
+     * Create TranslogCheckedContainer from provided bytes
      *
-     * @param channel {@link FileChannel} to read from
-     * @param offset  offset of channel from which bytes are to be read.
-     * @param len     Length of bytes to be read.
+     * @param bytes The byte array to read from
      */
-    public TranslogCheckedContainer(FileChannel channel, int offset, int len, String file) throws IOException {
```

Comment:
can we move checksum calculation into `BaseTranslogReader` itself, so that ckp and tlog file both can use it ?   Having just another container for checksum calculation doesn't look to add much value.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/8080